exchanges/okx/okx_websocket.go

```diff
@@ -0,0 +1,2155 @@
+		return nil, errInvalidOrderType
+	}
+	if arg.Amount <= 0 {
+		return nil, errInvalidAmount
+	}
+	if arg.Price <= 0 && (strings.EqualFold(arg.OrderType, "limit") || strings.EqualFold(arg.OrderType, "post_only") ||
+		strings.EqualFold(arg.OrderType, "fok") || strings.EqualFold(arg.OrderType, "ioc")) {
+		return nil, fmt.Errorf("invalid order price for %s order types", arg.OrderType)
+	}
+	if !(strings.EqualFold(arg.QuantityType, "base_ccy") || strings.EqualFold(arg.QuantityType, "quote_ccy")) {
+		arg.QuantityType = ""
+	}
+	randomID, err := common.GenerateRandomString(32, common.SmallLetters, common.CapitalLetters, common.NumberCharacters)
+	if err != nil {
+		return nil, err
+	}
+	input := WsPlaceOrderInput{
+		ID:        randomID,
+		Arguments: []PlaceOrderRequestParam{*arg},
+		Operation: okxOpOrder,
+	}
+	err = ok.Websocket.AuthConn.SendJSONMessage(input)
+	if err != nil {
+		return nil, err
+	}
+	timer := time.NewTimer(ok.WebsocketResponseMaxLimit)
+	wsResponse := make(chan *wsIncomingData)
+	ok.WsResponseMultiplexer.Register <- &wsRequestInfo{
+		ID:   randomID,
+		Chan: wsResponse,
+	}
+	defer func() { ok.WsResponseMultiplexer.Unregister <- randomID }()
+	for {
+		select {
+		case data := <-wsResponse:
+			if data.Operation == okxOpOrder && data.ID == input.ID {
+				if data.Code == "0" || data.Code == "1" {
+					resp, err := data.copyToPlaceOrderResponse()
+					if err != nil {
+						return nil, err
+					}
+					if len(resp.Data) != 1 {
+						return nil, errNoValidResponseFromServer
+					}
+					if data.Code == "1" {
+						return nil, fmt.Errorf("error code:%s message: %s", resp.Data[0].SCode, resp.Data[0].SMessage)
+					}
+					return &resp.Data[0], nil
+				}
+				return nil, fmt.Errorf("error code:%s message: %v", data.Code, ErrorCodes[data.Code])
+			}
+			continue
+		case <-timer.C:
+			timer.Stop()
+			return nil, fmt.Errorf("%s websocket connection: timeout waiting for response with an operation: %v",
+				ok.Name,
+				input.Operation)
+		}
+	}
+}
+
+// WsPlaceMultipleOrder creates an order through the websocket stream.
+func (ok *Okx) WsPlaceMultipleOrder(args []PlaceOrderRequestParam) ([]OrderData, error) {
+	for x := range args {
+		arg := args[x]
+		if arg.InstrumentID == "" {
+			return nil, errMissingInstrumentID
+		}
+		if !(TradeModeCross == arg.TradeMode ||
+			TradeModeIsolated == arg.TradeMode ||
+			TradeModeCash == arg.TradeMode) {
+			return nil, errInvalidTradeModeValue
+		}
+		if !(strings.EqualFold(arg.Side, "buy") || strings.EqualFold(arg.Side, "sell")) {
+			return nil, errMissingOrderSide
+		}
+		if !(strings.EqualFold(arg.OrderType, "market") || strings.EqualFold(arg.OrderType, "limit") || strings.EqualFold(arg.OrderType, "post_only") ||
+			strings.EqualFold(arg.OrderType, "fok") || strings.EqualFold(arg.OrderType, "ioc") || strings.EqualFold(arg.OrderType, "optimal_limit_ioc")) {
```

Comment:
```suggestion
		if !strings.EqualFold(arg.OrderType, "market") && !strings.EqualFold(arg.OrderType, "limit") && !strings.EqualFold(arg.OrderType, "post_only") && !strings.EqualFold(arg.OrderType, "fok") && !strings.EqualFold(arg.OrderType, "ioc") && !strings.EqualFold(arg.OrderType, "optimal_limit_ioc") {
```
Do these need to be equalfold? Do you not have consts for these already?